### PR TITLE
Context parameters

### PR DESCRIPTION
**Pull request by guyblink**
Added support for get, set and delete in python.
Exporting first level nested params in bash.

### CODE REVIEW

File: python/context.py [hunk 1]
```diff
@@ -49,3 +49,25 @@ def __resolve_inner_key(self, key, create_keys: bool = False):
...
+
+    def set(self, key, value):
+        path = key.split(".")
+        path = ["variables"] + path
+        result = self.internal_dict
+
+        for sub_key in path[:-1]:
+            result = result.setdefault(sub_key, {})
+
+        result[path[-1]] = value
```
Comment (itsofirblink) [inline python/context.py:70]: nonononono use the provided __set_item__ don't implement again :D

File: python/context.py [hunk 2]
```diff
@@ -49,3 +49,25 @@ def __resolve_inner_key(self, key, create_keys: bool = False):
...
         return current_item
+
+    def get(self, key):
+        path = key.split(".")
+        result = self.internal_dict
+
+        for sub_key in path:
+            result = result.get(sub_key, {})
+
+        return result
```
Comment (itsofirblink) [inline python/context.py:60]: same here :D

File: python/context.py [hunk 3]
```diff
@@ -49,3 +49,25 @@ def __resolve_inner_key(self, key, create_keys: bool = False):
...
+
+    def set(self, key, value):
+        path = key.split(".")
+        path = ["variables"] + path
```
Comment (itsofirblink) [inline python/context.py:64]: And if i start with `variables.`?

File: python/context.py [hunk 4]
```diff
@@ -49,3 +49,25 @@ def __resolve_inner_key(self, key, create_keys: bool = False):
...
+        result = self.internal_dict
+
+        for sub_key in path[:-1]:
+            result = result.setdefault(sub_key, {})
+
+        result[path[-1]] = value
+
+    def delete(self, key):
+        self.internal_dict["variables"].pop(key, None)
```
Comment (itsofirblink) [inline python/context.py:73]: And if start with `variables`?

**Review by itsofirblink — CHANGES_REQUESTED** [review]

File: python/context.py (hunk 4 above)
Comment (guyblink) [reply python/context.py:73]: Fixed

File: python/context.py (hunk 1 above)
Comment (guyblink) [reply python/context.py:70]: Fixed

File: python/context.py (hunk 3 above)
Comment (guyblink) [reply python/context.py:64]: Fixed

File: python/context.py (hunk 2 above)
Comment (guyblink) [reply python/context.py:60]: Fixed

**Review by itsofirblink — APPROVED** [review]

**Pull request merged**